### PR TITLE
Add sentry configuration and version management with poetry

### LINKED ISSUE #22: Add sentry_sdk

### PR DESCRIPTION
**Pull request by Ph0tonic**
Configure sentry for the service with flask extension and `tbump` to manage version.

Fixes #22 

Linked with https://github.com/cta-epfl/flux-jh/pull/15

### CODE REVIEW

**Comment by volodymyrss** [comment]
It's usually hard to understand what "proper" implies since it depends on many factors. If you want to change the version management, could you motivate it by referring to some standard?

**Comment by Ph0tonic** [comment]
So, what I mean by proper was to configure an automated way to bump a release version. So that there is no need to open every file to manually bump the version. I used the same tools as Jupyterhub use see https://github.com/your-tools/tbump.

**Comment by volodymyrss** [comment]
> So, what I meant by proper was to configure an automated way to bump a release version. 

Could you update the title to reflect that this is what you want to achieve?

> So that there is no need to open every file to manually bump the version. I used the same tools as Jupyterhub, see https://github.com/your-tools/tbump.

We use poetry here which has [built-in support](https://python-poetry.org/docs/cli/#version) for bumping package version. 

Bumping additional version record in files can be done with [a plugin](https://github.com/monim67/poetry-bumpversion) or [without](https://github.com/python-poetry/poetry/issues/144#issuecomment-1488038660) 
see [more discussion](https://github.com/python-poetry/poetry/issues/144).

**Comment by Ph0tonic** [comment]
I replace the written version in python files by using `importlib.metadata` however, I tried to make `poetry-bumpversion` work but doesn't seems to work correctly. So for now there is no way to generate a git commit as well as a git tag. Shoud I remove tbump ?

**Comment by volodymyrss** [comment]
> I replace the written version in python files by using `importlib.metadata` however, I tried to make `poetry-bumpversion` work but doesn't seems to work correctly. 

In what way does it not work?
You did not mention in disadvantages that it does not work correctly, or you mean the lack of git tag/commit?

Note that poetry developers [do not believe](https://github.com/python-poetry/poetry/issues/4971#issuecomment-1013930810) that it is wise to add multiple ways to track the version since it's better to have a single source of truth.

Also please change the issue title to avoid confusion, we are adding something somewhat complicated and dependent on choice, as you can see there are numerous ways to do this and many opinions.

> So for now there is no way to generate a git commit as well as a git tag. Shoud I remove tbump ?

**Comment by Ph0tonic** [comment]
So, I tested it afterward but the plugin correctly bump poetry version but do not generate the tag and commit.
Tbump also uses a single source of truth which is stored inside `pyproject.toml`.
To move forward, we can then first simple without any tbump and simply basic poetry and see later for generating a proper git tag and commit message automatically.

**Comment by volodymyrss** [comment]
> So, I tested it afterward but the plugin correctly bump poetry version but do not generate the tag and commit. 

> Tbump also uses a single source of truth which is stored inside `pyproject.toml`. 

as far as `tbump` is concerned, yes, but it is reasonably possible to end up with git tag mismatching record in the `pyproject.yaml` if something else is used, e.g. `poetry version` which seems natural since we are in fact using `poetry` here.

to reduce chance of issues we need https://github.com/cta-epfl/downloadservice/issues/24

> To move forward, we can then first simple without any tbump and simply basic poetry and see later for generating a proper git tag and commit message automatically.

Yes, in the action, upon PR merge.

**Comment by volodymyrss** [comment]
I meant, we could upgrade since it's new version (although no new features here) just that not 0.1.0 to 0.4.0, maybe to 0.1.1? (since it's backwards compatible, no features actually). but keeping 0.1.0 is good too.

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "downloadservice"
-version = "0.1.0"
+version = "0.4.0"
```
[the remark this answers is not in the document]
Comment (Ph0tonic) [reply pyproject.toml]: Change it back to 0.2.0 👍

**Review by volodymyrss — APPROVED** [review]

**Pull request merged**